### ISSUE TITLE
PosgreSQL12 compatibility 

### ISSUE THREAD

**Issue by volga629** [dreamfactorysoftware/dreamfactory#308]
Hello Everyone,
We trying get dreamfactory working with postgresql 12 and getting error of deprecated function.

```
 devdbmaster@10.142.0.7(32912):devdb [32352] ERROR:  column d.adsrc does not exist at character 72
```

PgSQL error

```
[root@dev1-pgsql log]# tail -f postgresql-2019-11-11-Mon.log
			AND relnamespace = (SELECT oid FROM pg_catalog.pg_namespace WHERE nspname = $2))
	ORDER BY a.attnum
2019-11-11 09:33:37 EST devdbmaster@10.142.0.7(32912):devdb [32352] ERROR:  column d.adsrc does not exist at character 72
2019-11-11 09:33:37 EST devdbmaster@10.142.0.7(32912):devdb [32352] STATEMENT:  SELECT a.attname, LOWER(format_type(a.atttypid, a.atttypmod)) AS type, d.adsrc, a.attnotnull, a.atthasdef,
		pg_catalog.col_description(a.attrelid, a.attnum) AS comment
	FROM pg_attribute a LEFT JOIN pg_attrdef d ON a.attrelid = d.adrelid AND a.attnum = d.adnum
	WHERE a.attnum > 0 AND NOT a.attisdropped
		AND a.attrelid = (SELECT oid FROM pg_catalog.pg_class WHERE relname=$1
			AND relnamespace = (SELECT oid FROM pg_catalog.pg_namespace WHERE nspname = $2))
	ORDER BY a.attnum

```

Release notes  postgresql12 

https://www.postgresql.org/docs/12/release-12.html

Remove obsolete pg_attrdef.adsrc column (Peter Eisentraut)

**Comment by tvarsis** [comment]
Any update on this or idea of when this will be fixed?

**Comment by u2men** [comment]
Here I'm posting my solution until it get fixed. 

1. Allow PG to add a column to system tables by adding allow_system_table_mods = 'on' inside postgresql.conf file.
2. Restart PG
3. Execute this alter statement on the target database
alter table pg_catalog.pg_attrdef
ADD COLUMN "adsrc" text;
--Optional steps
4. Remove allow_system_table_mods = 'on'  from postgresql.conf file.
5. Restart
-- end of Optional steps

Try again DreamFactory and you can get connected.

**Comment by wjgilmore** [comment]
Hi,

I'm pleased to report a new release including PostgreSQL compatibility will be available later this week.

**Comment by tvarsis** [comment]
Was this fixed in the 4.0 release? I can't see any release notes anywhere. Also, when are the docker images updated to latest version?

**Comment by wjgilmore** [comment]
Just a note DreamFactory now supports PostgreSQL 12 and PostgreSQL 13 will be supported in the 4.4 release which could be available as early as today!
